Sources/Transport/Streams/FoundationStream.swift

```diff
@@ -84,7 +84,7 @@
         // MARK: Connect
 
         public func connect() throws -> Stream {
-            if securityLayer == .tls {
+            if case .tls(_) = securityLayer {
                 upgradeSSL()
```

Comment:
Yeah, we should. I piped it to upgradeSSL now, but I'm not sure how to set these properties on a FoundationStream. I think this could be done in a separate PR, as it'll require investigation.